### PR TITLE
This PR addresses duplicated comments issue in GetComments.

### PR DESCRIPTION
**Pull request by samyakb**
There are some API changes introduced in this PR which is more easy than previous API. Now theres no need to call two methods of `GetComments` and `List` to get full list of comments.

There are some edge cases which are not yet solved for some threads. I will cover those edge cases in next PR. This will help to keep commits simple.

### CODE REVIEW

**Comment by samyakb** [comment]
Here it would be nice to squash earlier two commits in final commit.

File: comments.go [hunk 1]
```diff
@@ -150,292 +153,145 @@ func (r *Replies) UnmarshalJSON(b []byte) error {
...
-		"context": "100",
-		"sort":    sort,
-		"depth":   "8",
-	}
-	resp, err := c.client.Get(endpoint, opt)
+	resp, err := c.client.Get(endpoint, options)
 
 	if err != nil {
-		log.Fatal("Error in getting comments response")
+		log.Fatal("Error in getting response")
```
Comment (samyakb) [inline comments.go]: we can use `respError` method here

**Review by samyakb — COMMENTED** [review]
```
	// FIXME(hmble): This link is archived a good example for test
	// but somehow I couldn't get all comments from this thread. Maybe there is
	// some edge cases that I need to consider. As of now IMO issue is getting
	// parent article's more children array.

```
I would like to include this here. Just in case for future reference

**Review by samyakb — COMMENTED** [review]
LGTM
